ingestion/src/metadata/profiler/interface/sqlalchemy/databricks/profiler_interface.py

```diff
@@ -13,14 +13,74 @@
 Interfaces with database for all database engine
 supporting sqlalchemy abstraction layer
 """
+from typing import List
 
+from pyhive.sqlalchemy_hive import HiveCompiler
+from sqlalchemy import Column
 
+from metadata.generated.schema.entity.data.table import Column as OMColumn
+from metadata.generated.schema.entity.data.table import ColumnName, DataType
+from metadata.generated.schema.entity.services.databaseService import (
+    DatabaseServiceType,
+)
 from metadata.profiler.interface.sqlalchemy.profiler_interface import (
     SQAProfilerInterface,
 )
+from metadata.profiler.orm.converter.base import build_orm_col
 
 
 class DatabricksProfilerInterface(SQAProfilerInterface):
+    """Databricks profiler interface"""
+
+    def visit_column(self, *args, **kwargs):
+        result = super(  # pylint: disable=bad-super-call
+            HiveCompiler, self
+        ).visit_column(*args, **kwargs)
+        dot_count = result.count(".")
+        if dot_count > 2:
+            splitted_result = result.split(".", 2)[-1].split(".")
+            result = ".".join(result.split(".", 2)[:-1])
+            result += "." + "`.`".join(splitted_result)
+
+        return result
+
+    HiveCompiler.visit_column = visit_column
```

Comment:
why are we using the hive compiler here?

Reply:
Databricks extends HiveDIalect and uses it's compiler.
In order to support Struct, we had to override the functionality.

Reply:
thanks, added a comment.